php/utils.php

```diff
@@ -1495,14 +1495,18 @@ function past_tense_verb( $verb ) {
  * @return string
  */
 function get_php_binary() {
-	$wp_cli_php_used = getenv( 'WP_CLI_PHP_USED' );
-	if ( false !== $wp_cli_php_used ) {
-		return $wp_cli_php_used;
-	}
 
-	$wp_cli_php = getenv( 'WP_CLI_PHP' );
-	if ( false !== $wp_cli_php ) {
-		return $wp_cli_php;
+	// Non-PHAR installs do not PHP binary override.
+	if ( ! defined( 'WP_CLI_PHAR_PATH' ) || ! WP_CLI_PHAR_PATH ) {
+		$wp_cli_php_used = getenv( 'WP_CLI_PHP_USED' );
+		if ( false !== $wp_cli_php_used ) {
+			return $wp_cli_php_used;
+		}
+
+		$wp_cli_php = getenv( 'WP_CLI_PHP' );
+		if ( false !== $wp_cli_php ) {
+			return $wp_cli_php;
+		}
 	}
```

Comment:
It's a little bit difficult to follow this conditional...

Could we switch it to something like:

```php
// Phar installs always use PHP_BINARY
if ( inside_phar() ) {
    return PHP_BINARY;
}
// Otherwise, do the non-Phar logic.
```

The `WP_CLI\Utils\inside_phar()` function is a good one for determining whether it's a Phar.

Reply:
@danielbachhuber ah yea that's better, I didn't see that function.

I agree on the logic, it's probably better to have an extra return than nested conditionals.

How about the update?

Reply:
> How about the update?

Looks great! I love simple changes 😊